### ISSUE TITLE
Add possibility to configure remote machine name via command line parameters

### ISSUE THREAD

**Issue by ghostbuster91** [gojuno/mainframer#173]
A possible situation:
I have two machines for remote compilation but none of them is accessible from public network. One is in my home and the other in the office. 

To change the name I have to change the `config` file.
It would be easier to specify it just through command line arguments
 e.g. `./mainframer.sh -r REMOTE_MACHINE_NAME taskName`

In additions it would make intellij idea integration with plugin easier ;)

Such configuration might override configuration in `config` file.

**Comment by artem-zinnatullin** [comment]
Hm, but isn't it easier to change it just in the file and keep running Mainframer without any additional parameters?

**Comment by ghostbuster91** [comment]
But if I had such parameter I could create two separate command aliases or scripts.
With file I have to create script which will replace my remoteMachineName in mainframer configuration.

Ofc this parameter would be optional.

**Comment by artem-zinnatullin** [comment]
Ok I'm pretty open for such a parameter, @ming13 what do you think?

**Comment by arturdryomov** [comment]
We’ve discussed config and argument parity some time ago. At this time we used it on a single project using a single machine, so we shut it down. I guess we should rethink it in the light of plans to provide Mainframer as a system-wide tool.

**Comment by PaulWoitaschek** [comment]
+1 for this. I have a script I run as an alias for mainframer that writes the server I want to the config file and then calls mainframer.

When I'm home and developing from the couch it goes to my local machine but when I'm at a client I'm accessing that machine through the internet.

**Comment by artem-zinnatullin** [comment]
Ok, let's do it, I suppose we should support both short and long form of argument passing, i.e:

```
--remote-machine
-rm????
-r?

--local-compression-level
-lcl

--remote-compression-level
-rcl
```

**Comment by PaulWoitaschek** [comment]
Why not just something like:

`./mainframer -c configFile ./gradlew assembleRelease`

So you just point to a config file? When absent it uses the default config file.

**Comment by artem-zinnatullin** [comment]
Yup, that could also be added, however just passing config values as
arguments is valuable as well since it does not require managing files

**Comment by neworld** [comment]
+1. I am in the same situation, two remote machines in two different locations. All team is using mainframer and configs are in the repo. Sometimes I accidentally push these configs. It is quite annoying to fixing it.

**Comment by artem-zinnatullin** [comment]
@neworld for now you can temporary ignore changes you do in this file if you use git so you don't accidentally commit it http://gitready.com/intermediate/2009/02/18/temporarily-ignoring-files.html

But we heard you :)

**Comment by ghostbuster91** [comment]
Any updates on this?

**Comment by artem-zinnatullin** [comment]
Yep, I think this is important feature, let me add it to 3.x milestone

I'm also thinking about splitting `project/.mainframer/config` into two files, so one could be committed to the repo and the other one could be ignored, like `local.properties`

**Comment by Oleg-Beloy** [comment]
@artem-zinnatullin How is it going? As for us that's an urgent feature especially to speed up particular build configuration. For example for the app build we just need an apk but it wouldn't be enough for tests etc. 
At 2.0.0 version we don't have any other option except general ignore rules
